lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -103,7 +103,8 @@ CDataFrameAnalysisInstrumentation::CDataFrameAnalysisInstrumentation(const std::
 }
 
 void CDataFrameAnalysisInstrumentation::updateMemoryUsage(std::int64_t delta) {
-    std::int64_t memory{m_Memory.fetch_add(delta)};
+    m_Memory.fetch_add(delta);
+    std::int64_t memory{m_Memory.load()};
```

Comment:
Splitting it over two lines means the peak could be missed, if another thread added a negative delta in between the two lines.  I think the only way to make it genuinely atomic is:

```suggestion
    std::int64_t memory{m_Memory.fetch_add(delta) + delta};
```

It's a shame there isn't an `add_fetch` method.